x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/AuthorizationServiceTests.java

```diff
@@ -2780,21 +2782,19 @@ public void testAuthorizationEngineSelectionForCheckPrivileges() throws Exceptio
             }
             return null;
         }).when(engine).resolveAuthorizationInfo(any(Subject.class), anyActionListener());
-        AuthorizationEngine.PrivilegesCheckResult privilegesCheckResult = new AuthorizationEngine.PrivilegesCheckResult(
-            randomBoolean(),
-            Map.of(),
-            Map.of(),
-            Map.of()
+        AuthorizationEngine.PrivilegesCheckResult privilegesCheckResult = randomFrom(
+            ALL_CHECKS_SUCCESS_NO_DETAILS,
+            SOME_CHECKS_FAILURE_NO_DETAILS
         );
         doAnswer(i -> {
             assertThat(i.getArguments().length, equalTo(4));
             final Object arg1 = i.getArguments()[0];
             assertThat(arg1, instanceOf(AuthorizationInfo.class));
             AuthorizationInfo authorizationInfoArg = (AuthorizationInfo) arg1;
-            final Object arg4 = i.getArguments()[3];
-            assertThat(arg4, instanceOf(ActionListener.class));
+            final Object arg5 = i.getArguments()[3];
+            assertThat(arg5, instanceOf(ActionListener.class));
```

Comment:
Just curious: why does this have to be `arg5` now?

Reply:
It should be `arg4`.